apps/mobile/src/app/wallet/create-new-wallet.tsx

```diff
@@ -10,27 +10,43 @@ import { Button } from '@/components/button';
 import { MnemonicWordBox } from '@/components/create-new-wallet/mnemonic-word-box';
 import { TransText } from '@/components/trans-text';
 import { APP_ROUTES } from '@/constants';
+import { tempMnemonicStore } from '@/state/storage-persistors';
 import { useTheme } from '@shopify/restyle';
 import { BlurView } from 'expo-blur';
 import * as Clipboard from 'expo-clipboard';
 import { useRouter } from 'expo-router';
 
+import { generateMnemonic } from '@leather.io/crypto';
 import { Box, Text, Theme, TouchableOpacity } from '@leather.io/ui/native';
 
-function getTestMnemonic() {
-  if (process.env.EXPO_PUBLIC_SECRET_KEY) {
-    return process.env.EXPO_PUBLIC_SECRET_KEY;
+function MnemonicDisplay({ mnemonic }: { mnemonic: string | null }) {
+  if (!mnemonic) {
+    return null;
   }
-  throw new Error('No mnemonic is provided in EXPO_PUBLIC_SECRET_KEY');
+  const mnemonicArr = mnemonic.split(' ');
+  return (
+    <Box justifyContent="center" flexDirection="row" flexWrap="wrap" gap="2">
+      {mnemonicArr.map((word, idx) => (
+        <MnemonicWordBox key={word + idx} wordIdx={idx + 1} word={word} />
+      ))}
+    </Box>
+  );
 }
 
 export default function CreateNewWallet() {
   const { bottom } = useSafeAreaInsets();
   const theme = useTheme<Theme>();
   const router = useRouter();
-  const test_array = getTestMnemonic().split(' ');
   const [isHidden, setIsHidden] = useState(true);
   const [isCopied, setIsCopied] = useState(false);
+  const [mnemonic, setMnemonic] = useState<string | null>(null);
+
+  useEffect(() => {
+    const tempMnemonic = generateMnemonic();
+    setMnemonic(tempMnemonic);
+    tempMnemonicStore.setTemporaryMnemonic(tempMnemonic);
```

Comment:
I can wire this up to store separately

Reply:
Yeah makes sense, just need to make sure this is the mnemonic that gets added to the store